### PR TITLE
[KOGITO-5376] - Adding deprecated addons from RHPAM 7.11 release

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

See: https://issues.redhat.com/browse/KOGITO-5376

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-build/kogito-ide-config)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Review by cristianonicolai — APPROVED** [review]

**Comment by ricardozanini** [comment]
Jenkins run apps tests

**Comment by ricardozanini** [comment]
Jenkins run apps tests

File: addons/deprecated/kogito-event-driven-decisions-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/events/deprecated/DeprecatedQuarkusEventDrivenDecisions.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.runtime.Startup;
+
+/**
+ * @deprecated The library kogito-event-driven-decisions-quarkus-addon is deprecated!!!!! Please use kogito-addons-quarkus-events-decisions instead
```
Comment (danielezonca) [inline addons/deprecated/kogito-event-driven-decisions-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/events/deprecated/DeprecatedQuarkusEventDrivenDecisions.java:26]: One exclamation mark is enough :) 
```suggestion
 * @deprecated The library kogito-event-driven-decisions-quarkus-addon is deprecated and will be removed in a future release! Please use kogito-addons-quarkus-events-decisions instead
```

File: addons/deprecated/kogito-event-driven-decisions-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/events/deprecated/DeprecatedQuarkusEventDrivenDecisions.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+@Startup
+@ApplicationScoped
+@Deprecated
+public class DeprecatedQuarkusEventDrivenDecisions {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedQuarkusEventDrivenDecisions.class);
+
+    public DeprecatedQuarkusEventDrivenDecisions() {
+        LOGGER.warn("The library kogito-event-driven-decisions-quarkus-addon is deprecated!!!!! Please use kogito-addons-quarkus-events-decisions instead");
```
Comment (danielezonca) [inline addons/deprecated/kogito-event-driven-decisions-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/events/deprecated/DeprecatedQuarkusEventDrivenDecisions.java:36]: Same as above
```suggestion
        LOGGER.warn("The library kogito-event-driven-decisions-quarkus-addon is deprecated and will be removed in a future release! Please use kogito-addons-quarkus-events-decisions instead");
```

File: addons/deprecated/kogito-event-driven-decisions-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/events/deprecated/DeprecatedSpringBootEvents.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * @deprecated The library kogito-event-driven-decisions-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-events-decisions instead
```
Comment (danielezonca) [inline addons/deprecated/kogito-event-driven-decisions-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/events/deprecated/DeprecatedSpringBootEvents.java:26]: Same comment

File: addons/deprecated/kogito-event-driven-decisions-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/events/deprecated/DeprecatedSpringBootEvents.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedSpringBootEvents.class);
+
+    public DeprecatedSpringBootEvents() {
+        LOGGER.warn("The library kogito-event-driven-decisions-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-events-decisions instead");
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        LOGGER.warn("The library kogito-event-driven-decisions-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-events-decisions instead");
```
Comment (danielezonca) [inline addons/deprecated/kogito-event-driven-decisions-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/events/deprecated/DeprecatedSpringBootEvents.java:41]: Why are you logging twice?
Same comment about `!!!!!`

File: addons/deprecated/monitoring-core-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/monitoring/deprecated/DeprecatedQuarkusMonitoringCore.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ */
+@Startup
+@ApplicationScoped
+@Deprecated
+public class DeprecatedQuarkusMonitoringCore {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedQuarkusMonitoringCore.class);
+
+    public DeprecatedQuarkusMonitoringCore() {
+        LOGGER.warn("The library monitoring-core-quarkus-addon is deprecated!!!!! Please use kogito-addons-quarkus-monitoring-core instead");
```
Comment (danielezonca) [inline addons/deprecated/monitoring-core-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/monitoring/deprecated/DeprecatedQuarkusMonitoringCore.java:36]: Same comment

File: addons/deprecated/monitoring-core-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/monitoring/deprecated/DeprecatedQuarkusMonitoringCore.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.runtime.Startup;
+
+/**
+ * @deprecated The library monitoring-core-quarkus-addon is deprecated!!!!! Please use kogito-addons-quarkus-monitoring-core instead
```
Comment (danielezonca) [inline addons/deprecated/monitoring-core-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/monitoring/deprecated/DeprecatedQuarkusMonitoringCore.java:26]: Same comment

File: addons/deprecated/monitoring-core-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/monitoring/deprecated/DeprecatedSpringBootMonitoringCore.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedSpringBootMonitoringCore.class);
+
+    public DeprecatedSpringBootMonitoringCore() {
+        LOGGER.warn("The library monitoring-core-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-monitoring-core instead");
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        LOGGER.warn("The library monitoring-core-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-monitoring-core instead");
```
Comment (danielezonca) [inline addons/deprecated/monitoring-core-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/monitoring/deprecated/DeprecatedSpringBootMonitoringCore.java:41]: Same comment

File: addons/deprecated/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/cloudevents/deprecated/DeprecatedQuarkusCloudEvents.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+ */
+@Startup
+@ApplicationScoped
+@Deprecated
+public class DeprecatedQuarkusCloudEvents {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedQuarkusCloudEvents.class);
+
+    public DeprecatedQuarkusCloudEvents() {
+        LOGGER.warn("The library cloudevents-quarkus-addon is deprecated!!!!! Please use kogito-addons-quarkus-cloudevents instead");
```
Comment (danielezonca) [inline addons/deprecated/cloudevents-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/cloudevents/deprecated/DeprecatedQuarkusCloudEvents.java:36]: Same comment

File: addons/deprecated/cloudevents-spring-boot-addon/src/main/java/org/kie/kogito/addons/springboot/cloudevents/deprecated/DeprecatedSpringBootCloudEvents.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedSpringBootCloudEvents.class);
+
+    public DeprecatedSpringBootCloudEvents() {
+        LOGGER.warn("The library cloudevents-spring-boot-addon is deprecated!!!!! Please use kogito-addons-springboot-cloudevents instead");
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        LOGGER.warn("The library cloudevents-spring-boot-addon is deprecated!!!!! Please use kogito-addons-springboot-cloudevents instead");
```
Comment (danielezonca) [inline addons/deprecated/cloudevents-spring-boot-addon/src/main/java/org/kie/kogito/addons/springboot/cloudevents/deprecated/DeprecatedSpringBootCloudEvents.java:41]: Same comment

File: addons/deprecated/knative-eventing-addon/src/main/java/org/kie/kogito/addons/quarkus/knative/deprecated/DeprecatedQuarkusKnative.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+ */
+@Startup
+@ApplicationScoped
+@Deprecated
+public class DeprecatedQuarkusKnative {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedQuarkusKnative.class);
+
+    public DeprecatedQuarkusKnative() {
+        LOGGER.warn("The library knative-eventing-addon is deprecated!!!!! Please use please use kogito-addons-quarkus-cloudevents and io.quarkus:quarkus-reactive-messaging-http instead");
```
Comment (danielezonca) [inline addons/deprecated/knative-eventing-addon/src/main/java/org/kie/kogito/addons/quarkus/knative/deprecated/DeprecatedQuarkusKnative.java:36]: Same comment

File: addons/deprecated/monitoring-core-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/monitoring/prometheus/deprecated/DeprecatedSpringBootMonitoringPrometheus.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedSpringBootMonitoringPrometheus.class);
+
+    public DeprecatedSpringBootMonitoringPrometheus() {
+        LOGGER.warn("The library monitoring-prometheus-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-monitoring-prometheus instead");
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        LOGGER.warn("The library monitoring-prometheus-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-monitoring-prometheus instead");
+    }
```
Comment (danielezonca) [inline addons/deprecated/monitoring-core-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/monitoring/prometheus/deprecated/DeprecatedSpringBootMonitoringPrometheus.java:42]: Same comment

File: addons/deprecated/monitoring-elastic-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/monitoring/elastic/deprecated/DeprecatedQuarkusMonitoringElastic.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+ */
+@Startup
+@ApplicationScoped
+@Deprecated
+public class DeprecatedQuarkusMonitoringElastic {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedQuarkusMonitoringElastic.class);
+
+    public DeprecatedQuarkusMonitoringElastic() {
+        LOGGER.warn("The library monitoring-elastic-quarkus-addon is deprecated!!!!! Please use kogito-addons-quarkus-monitoring-elastic instead");
```
Comment (danielezonca) [inline addons/deprecated/monitoring-elastic-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/monitoring/elastic/deprecated/DeprecatedQuarkusMonitoringElastic.java:36]: Same comment

File: addons/deprecated/monitoring-elastic-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/monitoring/elastic/deprecated/DeprecatedSpringBootMonitoringElastic.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedSpringBootMonitoringElastic.class);
+
+    public DeprecatedSpringBootMonitoringElastic() {
+        LOGGER.warn("The library monitoring-elastic-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-monitoring-elastic instead");
+    }
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        LOGGER.warn("The library monitoring-elastic-springboot-addon is deprecated!!!!! Please use kogito-addons-springboot-monitoring-elastic instead");
```
Comment (danielezonca) [inline addons/deprecated/monitoring-elastic-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/monitoring/elastic/deprecated/DeprecatedSpringBootMonitoringElastic.java:41]: Same comment

File: addons/deprecated/monitoring-prometheus-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/monitoring/prometheus/deprecated/DeprecatedQuarkusMonitoringPrometheus.java [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+ */
+@Startup
+@ApplicationScoped
+@Deprecated
+public class DeprecatedQuarkusMonitoringPrometheus {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedQuarkusMonitoringPrometheus.class);
+
+    public DeprecatedQuarkusMonitoringPrometheus() {
+        LOGGER.warn("The library monitoring-prometheus-quarkus-addon is deprecated!!!!! Please use kogito-addons-quarkus-monitoring-prometheus instead");
```
Comment (danielezonca) [inline addons/deprecated/monitoring-prometheus-quarkus-addon/src/main/java/org/kie/kogito/addons/quarkus/monitoring/prometheus/deprecated/DeprecatedQuarkusMonitoringPrometheus.java:36]: Same comment

File: addons/deprecated/pom.xml [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+    <module>kogito-event-driven-decisions-common</module>
+    <module>kogito-event-driven-decisions-quarkus-addon</module>
+    <module>kogito-event-driven-decisions-springboot-addon</module>
+    <module>knative-eventing-addon</module>
+    <module>monitoring-core-common</module>
+    <module>monitoring-core-quarkus-addon</module>
+    <module>monitoring-core-springboot-addon</module>
+    <module>monitoring-elastic-common</module>
+    <module>monitoring-elastic-quarkus-addon</module>
+    <module>monitoring-elastic-springboot-addon</module>
```
Comment (danielezonca) [inline addons/deprecated/pom.xml:30]: I think it is safe to remove `elastic` one. It has been never supported and it is just an experiment to show how to use a different technology

File: addons/deprecated/kogito-event-driven-decisions-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/events/deprecated/DeprecatedSpringBootEvents.java (hunk 4 above)
Comment (ricardozanini) [reply addons/deprecated/kogito-event-driven-decisions-springboot-addon/src/main/java/org/kie/kogito/addons/springboot/events/deprecated/DeprecatedSpringBootEvents.java:41]: 'cause one of them might pass unnoticed (there's a lot of logs on SB boot), let it there :)
Hopefully, they will be bugged enough to migrate. I'll change the message, though.

**Review by danielezonca — APPROVED** [review]

File: addons/deprecated/cloudevents-common-addon/pom.xml [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>kogito-addons-deprecated-parent</artifactId>
+    <groupId>org.kie.kogito</groupId>
+    <version>2.0.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>cloudevents-common-addon</artifactId>
```
Comment (MarianMacik) [inline addons/deprecated/cloudevents-common-addon/pom.xml:12]: Shouldn't this be called the old way, i.e. `kogito-cloudevents-common-addon`?

File: addons/deprecated/cloudevents-quarkus-addon/pom.xml [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>kogito-addons-deprecated-parent</artifactId>
+    <groupId>org.kie.kogito</groupId>
+    <version>2.0.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>cloudevents-quarkus-addon</artifactId>
```
Comment (MarianMacik) [inline addons/deprecated/cloudevents-quarkus-addon/pom.xml:12]: Same here, `kogito-cloudevents-quarkus-addon`.

File: addons/deprecated/cloudevents-spring-boot-addon/pom.xml [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>kogito-addons-deprecated-parent</artifactId>
+    <groupId>org.kie.kogito</groupId>
+    <version>2.0.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>cloudevents-spring-boot-addon</artifactId>
```
Comment (MarianMacik) [inline addons/deprecated/cloudevents-spring-boot-addon/pom.xml:12]: kogito-cloudevents-spring-boot-addon

File: addons/deprecated/knative-eventing-addon/src/main/java/org/kie/kogito/addons/quarkus/knative/deprecated/DeprecatedQuarkusKnative.java [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.runtime.Startup;
+
+/**
+ * @deprecated The library knative-eventing-addon is deprecated and will be removed in a future release! Please use please use kogito-addons-quarkus-cloudevents and
```
Comment (MarianMacik) [inline addons/deprecated/knative-eventing-addon/src/main/java/org/kie/kogito/addons/quarkus/knative/deprecated/DeprecatedQuarkusKnative.java:26]: We don't have to ask twice :)
```suggestion
 * @deprecated The library knative-eventing-addon is deprecated and will be removed in a future release! Please use kogito-addons-quarkus-cloudevents and
```

File: addons/deprecated/knative-eventing-addon/src/main/java/org/kie/kogito/addons/quarkus/knative/deprecated/DeprecatedQuarkusKnative.java [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+@Startup
+@ApplicationScoped
+@Deprecated
+public class DeprecatedQuarkusKnative {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(DeprecatedQuarkusKnative.class);
+
+    public DeprecatedQuarkusKnative() {
+        LOGGER.warn(
+                "The library knative-eventing-addon is deprecated and will be removed in a future release! Please use please use kogito-addons-quarkus-cloudevents and io.quarkus:quarkus-reactive-messaging-http instead");
```
Comment (MarianMacik) [inline addons/deprecated/knative-eventing-addon/src/main/java/org/kie/kogito/addons/quarkus/knative/deprecated/DeprecatedQuarkusKnative.java:38]: Same here.

File: addons/deprecated/cloudevents-common-addon/pom.xml (hunk 16 above)
Comment (ricardozanini) [reply addons/deprecated/cloudevents-common-addon/pom.xml:12]: Dammit, I was misled by the folder name. :facepalm:

**Review by MarianMacik — APPROVED** [review]

**Comment by ricardozanini** [comment]
Jenkins retest this

**Pull request merged**